azplugins/flow.py

```diff
@@ -717,25 +722,41 @@ def __call__(self, timestep):
 
         if hoomd.comm.get_rank() == 0:
             x = snap.particles.position[:,self.bin_axis]
-            v = snap.particles.velocity[:,self.flow_axis]
+            v = snap.particles.velocity
+            m = snap.particles.mass
 
-            _counts,_ = np.histogram(x, bins=self.bins, range=self.range)
-            self._counts += _counts
+            binids = np.digitize(x, self.edges[1:])
 
-            _velocity,_ = np.histogram(x, bins=self.bins, range=self.range, weights=v)
-            self._velocity += _velocity
+            for dim in range(3):
+               num_vel = np.bincount(binids,v[:,dim],minlength=self.bins)
+               counts = np.bincount(binids,minlength=self.bins)
+               self._mass_velocity[dim] += np.bincount(binids,m*v[:,dim],minlength=self.bins)
+               self._number_velocity[dim] += np.divide(num_vel, counts, out=np.zeros(self.bins), where=counts > 0)
 
+            vsq = np.sum(v**2,axis=1)
+            vsqm = np.bincount(binids, weights=vsq*m, minlength=self.bins)
+            self._vsqm +=vsqm
```

Comment:
Now for this calculation, we can subtract the center-of-mass velocity from `mass_vel`. I think there's a way to do this just correcting the kinetic energy of the particles (without this velocity subtracted). The formulas are in the MPCD temperature compute code... I can look for them if you want me to.

Reply:
Please double check, I think the temperature is correct now.